### DISCUSSION TITLE
Some values not received on mqtt

### DISCUSSION THREAD

**Discussion by perico85** [johanmeijer/grott#253] [Q&A]
Hello, first of all, thank you for the tool.

I have a growatt min5000tl inverter that communicates to the growatt server with a shinelan X. I have it configured in home assistant through the "egguy" utility. The problem I have is that it does not give me values such as battery charge, imported from the mains or exported.

I send log of received mqtt messages.

> 
- Grott values retrieved:
		 -  datalogserial        :  KWK1XXYYZZ
		 -  pvserial             :  YBK4CFHXYZ
		 -  pvstatus             :  1
		 -  pvpowerin            :  903.1
		 -  pv1voltage           :  120.2
		 -  pv1current           :  3.2
		 -  pv1watt              :  395.3
		 -  pv2voltage           :  194.7
		 -  pv2current           :  2.5
		 -  pv2watt              :  507.8
		 -  pvpowerout           :  0.0
		 -  pvfrequentie         :  0.0
		 -  pvgridvoltage        :  219.1
		 -  pvgridcurrent        :  0.0
		 -  pvgridpower          :  0.0
		 -  pvgridvoltage2       :  0.0
		 -  pvgridcurrent2       :  0.0
		 -  pvgridpower2         :  0.0
		 -  pvgridvoltage3       :  420.1
		 -  pvgridcurrent3       :  10.1
		 -  pvgridpower3         :  403891814.4
		 -  totworktime          :  0.4
		 -  eactoday             :  646.3
		 -  pvenergytoday        :  646.3
		 -  eactotal             :  2.5
		 -  epvtotal             :  0.0
		 -  epv1today            :  4.8
		 -  epv1total            :  368.5
		 -  epv2today            :  0.0
		 -  epv2total            :  0.0
		 -  pvtemperature        :  34.0
		 -  pvipmtemperature     :  34.0
		 -  pvboosttemp          :  0.0
		 -  bat_dsp              :  38.7
		 -  eacharge_today       :  0.0
		 -  eacharge_total       :  18.0
		 -  batterytype          :  0
		 -  uwsysworkmode        :  0
		 -  systemfaultword0     :  20
		 -  systemfaultword1     :  0
		 -  systemfaultword2     :  1927
		 -  systemfaultword3     :  0
		 -  systemfaultword4     :  30
		 -  systemfaultword5     :  0
		 -  systemfaultword6     :  2125
		 -  systemfaultword7     :  0
		 -  pdischarge1          :  0.0
		 -  p1charge1            :  0.0
		 -  vbat                 :  627.8
		 -  SOC                  :  0.0
		 -  pactouserr           :  412876.8
		 -  pactousertot         :  0.0
		 -  pactogridr           :  0.0
		 -  pactogridtot         :  0.0
		 -  plocaloadr           :  0.0
		 -  plocaloadtot         :  390.0
		 -  spdspstatus          :  25.7
		 -  spbusvolt            :  0.0
		 -  etouser_tod          :  66355204.7
		 -  etouser_tot          :  190444.4
		 -  etogrid_tod          :  13153077.5
		 -  etogrid_tot          :  150764.1
		 -  edischarge1_tod      :  0.0
		 -  edischarge1_tot      :  0.0
		 -  eharge1_tod          :  31391744.0
		 -  eharge1_tot          :  12628787.2
		 -  elocalload_tod       :  13926400.0
		 -  elocalload_tot       :  19850.1
	 - MQTT jsonmsg: 
			 {"device": "YBK4CFHXYZ", "time": "2022-12-06T14:37:41", "buffered": "no",
			 "values": {"datalogserial": "KWK1XXYYZZ", "pvserial": "YBK4CFH034",
			 "pvstatus": 1, "pvpowerin": 9031, "pv1voltage": 1202, "pv1current": 32,
			 "pv1watt": 3953, "pv2voltage": 1947, "pv2current": 25, "pv2watt": 5078,
			 "pvpowerout": 0, "pvfrequentie": 0, "pvgridvoltage": 2191, "pvgridcurrent":
			 0, "pvgridpower": 0, "pvgridvoltage2": 0, "pvgridcurrent2": 0,
			 "pvgridpower2": 0, "pvgridvoltage3": 4201, "pvgridcurrent3": 101,
			 "pvgridpower3": 4038918144, "totworktime": 2778, "eactoday": 6463,
			 "pvenergytoday": 6463, "eactotal": 25, "epvtotal": 0, "epv1today": 48,
			 "epv1total": 3685, "epv2today": 0, "epv2total": 0, "pvtemperature": 340,
			 "pvipmtemperature": 340, "pvboosttemp": 0, "bat_dsp": 387, "eacharge_today":
			 0, "eacharge_total": 180, "batterytype": 0, "uwsysworkmode": 0,
			 "systemfaultword0": 20, "systemfaultword1": 0, "systemfaultword2": 1927,
			 "systemfaultword3": 0, "systemfaultword4": 30, "systemfaultword5": 0,
			 "systemfaultword6": 2125, "systemfaultword7": 0, "pdischarge1": 0,
			 "p1charge1": 0, "vbat": 6278, "SOC": 0, "pactouserr": 4128768,
			 "pactousertot": 0, "pactogridr": 0, "pactogridtot": 0, "plocaloadr": 0,
			 "plocaloadtot": 3900, "spdspstatus": 257, "spbusvolt": 0, "etouser_tod":
			 663552047, "etouser_tot": 1904444, "etogrid_tod": 131530775, "etogrid_tot":
			 1507641, "edischarge1_tod": 0, "edischarge1_tot": 0, "eharge1_tod":
			 313917440, "eharge1_tot": 126287872, "elocalload_tod": 139264000,
			 "elocalload_tot": 198501}}
	 - Grott MQTT topic used : energy/growatt
	 - MQTT message message sent
	 - Grott Send data to PVOutput disabled 
	 - Grott Send data to Influx disabled 
	 - Grott extension processing started :  grott_ha
	Grott HA - creating YBK4CFHXYZ config in HA
	 - Grott extension processing ended :  0

**Comment by johanmeijer** [comment]
Hmm I am not sure we already have the right layout for a min type inverter. Have you specified inverter =  in the .ini ant what type do you use? 

Is it a min 5000 tl-x?

**Reply by perico85** [reply]
yes, it's a "min 5000tl-x".


![Screenshot_20221206_231004_com android chrome_edit_259323981415638](https://user-images.githubusercontent.com/4411094/206034976-3304c290-d9a5-455c-aea8-a5effb70bba3.jpg)
![Screenshot_20221206_231028_com android chrome_edit_259437329489579](https://user-images.githubusercontent.com/4411094/206034998-d71bb6dd-3aeb-42f4-9962-f1102b68fbcf.jpg)


 in inverter type option, i have used "sph".

**Comment by egguy** [comment]
Hi,

In the add-on, the inverter's configuration variable is `ginvtype`, If your inverter is a min5000, the value SPH seems inappropriate, this value is for the SPHX000 series (ex: SPH-6000). This could explain the missing information.

I also don't have 100% of the mappings done (ex: `epv1today` -> PV1 solar production (Today)), but if you have a list of the missing mapping and their descriptions / unit, I can add the new mappings quickly.

If you use the raw MQTT output, then you have to do your configuration, there's an example in `examples/Home Assistent/sensors_growatt_eng.yaml`

**Reply by perico85** [reply]
thanks for your answer, i will look for the values that are missing with the johanmeijer xls documento.
